### PR TITLE
Port SimpleScenariosForLocalDb to .NET Core 3

### PR DESCRIPTION
**Pull request by bricelam**
Note, this is not a pattern for testing on .NET Core--just a quick and dirty way to get basic coverage for now.

Part of #271

### CODE REVIEW

File: test/EntityFramework/FunctionalTests.netcoreapp3.0/ProductivityApi/SimpleScenariosForLocalDb.cs [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    public class SimpleScenariosForLocalDb : FunctionalTestBase, IDisposable
+    {
+        #region Infrastructure/setup
+
+        private readonly object _previousDataDirectory;
+
+        public SimpleScenariosForLocalDb()
+        {
+            // TODO: Can we hardcode this somewhere like we do for the EF provider?
+            DbProviderFactories.RegisterFactory("System.Data.SqlClient", SqlClientFactory.Instance);
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests.netcoreapp3.0/ProductivityApi/SimpleScenariosForLocalDb.cs]: All the test files were just coppied and trimmed to make them compile.

This is the only line added. Users will need to put this in their app startup for Preview 5.

File: test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <PackageReference Include="xunit.assert" Version="$(XUnitVersion)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\src\EntityFramework.SqlServer\EntityFramework.SqlServer.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!-- HACK: xUnit.net uses AppDomains on .NET Framework to make App.config work -->
+    <None Update="xunit.console.dll.config">
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj]: ☹️ This only makes `build.cmd` work. Test will fail in other test runners (including Visual Studio)

File: test/EntityFramework/FunctionalTests.netcoreapp3.0/ProductivityApi/SimpleScenariosForLocalDb.cs (hunk 1 above)
Comment (divega) [reply test/EntityFramework/FunctionalTests.netcoreapp3.0/ProductivityApi/SimpleScenariosForLocalDb.cs]: I guess we could add fallback logic every time we ask for the factory.

Comment (bricelam) [reply test/EntityFramework/FunctionalTests.netcoreapp3.0/ProductivityApi/SimpleScenariosForLocalDb.cs]: I think we already have the indirection; I just didn’t want to rush the change. I’ll investigate Monday morning to see if there’s a simple change we can make to Preview 5.

File: test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj (hunk 2 above)
Comment (divega) [reply test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj]: What is the underlying issue? Is this related to the fact that a lot of functionality in EF6 still depends on entries in the config file? Assumig that is the case, will the tests work when/if we add sensible default behaviors? Or should the tests change to do code-based configuration?

Comment (divega) [reply test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj]: BTW, I am asking about what happens after preview5. I think tis is fine for now.

Comment (bricelam) [reply test/EntityFramework/FunctionalTests.netcoreapp3.0/FunctionalTests.netcoreapp3.0.csproj]: I still need to investigate alternatives. There may be a way to change where ConfigurationManager looks...

**Pull request merged**